File: Simulador_de_dado.py
#Simulando o uso de dados de 1 até 6
import random

#Definindo parametros
class SimuladorDeDado:
    def __init__(self):
        self.valor_minimo: 1
        self.valor_maximo: 6
        self.mensagem = 'Você gostaria de gerar um valor para os dados?'

#Recebendo comando
    def Iniciar(self):
        resposta = input(self.mensagem)
        try:
#Inicializando            
            if resposta == 'sim':
                self.GerarValorDoDado()
            elif resposta == 'não':
                print('Volte sempre!')
            else:
                print('Digite sim ou não')
        except:
            print('Ocorreu um erro em sua resposta')


    def GerarValorDoDado(self):
        print(random.randint(self.valor_minimo,self.valor_maximo))

simulador = SimuladorDeDado()
simulador.Iniciar()

File: Simulador_de_dado.rb
#Simulando o uso de dados de 1 até 6
require 'securerandom'

#Definindo parametros
class SimuladorDeDados
  def initialize
    @valor_minimo = 1
    @valor_maximo = 6
    @mensagem = 'Você gostaria de gerar um valor para os dados?'
  end

  #Recebendo comando
  def iniciar
    resposta = gets.chomp.downcase
    begin
      #Inicializando
      if resposta == 'sim'
        gerar_valor_do_dado
      elsif
        puts 'Volte sempre!'
      else
        puts 'Digite sim ou não'
      end
    rescue StandError
      puts 'Ocorreu um erro em sua resposta'
    end
  end

  def gerar_valor_do_dado
    puts SecureRandom.random_number(@valor_maximo - @valor_minimo + 1) + valor_minimo
  end
end

simulador = SimuladorDeDados.new
puts simulador.mensagem
simulador.iniciar
